state/action.go

```diff
@@ -437,7 +437,8 @@ func (a *action) removeAndLogBuildTxn(finalStatus ActionStatus, results map[stri
 					numComplete++
 				}
 			}
-			if numComplete == len(tasks)-1 {
+			expectedTaskCount := parentOperation.(*operation).doc.ExpectedTaskCount
```

Comment:
Can you please add a type check for the cast to `*operation` just to be on the safe side?